firebase-firestore/src/androidTest/java/com/google/firebase/firestore/AggregationTest.java

```diff
@@ -0,0 +1,1030 @@
+    AggregateField f2 = average("pages");
+    AggregateField f3 = AggregateField.count();
+    AggregateField f4 = sum("foo");
+    AggregateField f5 = sum("bar");
+
+    AggregateQuerySnapshot snapshot =
+        waitFor(collection.aggregate(f1, f2, f3, f4, f5).get(AggregateSource.SERVER));
+
+    assertEquals(snapshot.get(f1), 150L);
+    assertEquals(snapshot.get(f2), 75.0);
+    assertEquals(snapshot.get(f3), 2L);
+    assertEquals(snapshot.get(f4), 2L);
+    assertEquals(snapshot.get(f5), 4L);
+  }
+
+  @Test
+  public void testCannotPerformMoreThanMaxAggregations() {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+    AggregateField f1 = sum("pages");
+    AggregateField f2 = average("pages");
+    AggregateField f3 = AggregateField.count();
+    AggregateField f4 = sum("foo");
+    AggregateField f5 = sum("bar");
+    AggregateField f6 = sum("baz");
+
+    Exception e =
+        waitForException(collection.aggregate(f1, f2, f3, f4, f5, f6).get(AggregateSource.SERVER));
+
+    assertThat(e, instanceOf(FirebaseFirestoreException.class));
+    FirebaseFirestoreException firestoreException = (FirebaseFirestoreException) e;
+    assertEquals(FirebaseFirestoreException.Code.INVALID_ARGUMENT, firestoreException.getCode());
+    assertTrue(firestoreException.getMessage().contains("maximum number of aggregations"));
+  }
+
+  @Test
+  public void testCanRunCountCollectionGroupQuery() {
+    FirebaseFirestore db = testFirestore();
+    // Use .document() to get a random collection group name to use but ensure it starts with 'b'
+    // for predictable ordering.
+    String collectionGroup = "b" + db.collection("foo").document().getId();
+
+    String[] docPaths =
+        new String[] {
+          "abc/123/${collectionGroup}/cg-doc1",
+          "abc/123/${collectionGroup}/cg-doc2",
+          "${collectionGroup}/cg-doc3",
+          "${collectionGroup}/cg-doc4",
+          "def/456/${collectionGroup}/cg-doc5",
+          "${collectionGroup}/virtual-doc/nested-coll/not-cg-doc",
+          "x${collectionGroup}/not-cg-doc",
+          "${collectionGroup}x/not-cg-doc",
+          "abc/123/${collectionGroup}x/not-cg-doc",
+          "abc/123/x${collectionGroup}/not-cg-doc",
+          "abc/${collectionGroup}"
+        };
+    WriteBatch batch = db.batch();
+    for (String path : docPaths) {
+      batch.set(db.document(path.replace("${collectionGroup}", collectionGroup)), map("x", 2));
+    }
+    waitFor(batch.commit());
+
+    AggregateQuerySnapshot snapshot =
+        waitFor(
+            db.collectionGroup(collectionGroup)
+                .aggregate(AggregateField.count(), sum("x"), average("x"))
+                .get(AggregateSource.SERVER));
+    assertEquals(
+        5L, // "cg-doc1", "cg-doc2", "cg-doc3", "cg-doc4", "cg-doc5",
+        snapshot.get(AggregateField.count()));
+    assertEquals(10L, snapshot.get(sum("x")));
+    assertEquals((Double) 2.0, snapshot.get(average("x")));
+  }
+
```

Comment:
Might have missed the test [performsAggregationsOnDocumentsWithAllaggregatedFields](https://github.com/firebase/firebase-js-sdk/blob/83740c08c0597c3f47ccbeb9afec6ebfa7ad753a/packages/firestore/test/integration/api/aggregation.test.ts#L465)

Reply:
I think the test would be the same as `testCanGetMultipleAggregationsInTheSameQuery`?